### PR TITLE
[TASK] Adding more logs for ramp feature + fixing a bug on saving ramp status to DB

### PR DESCRIPTION
**Pull request by kxu1026**
ADD MORE LOGS + FIX ONE BUG WHICH CAUSE THE CACHED STATUS CANNOT BE SAVED INTO DB

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java [hunk 1]
```diff
@@ -365,12 +369,27 @@ synchronized public void cacheResult(Action action) {
             / ((this.state.numOfTrail + this.state.cachedNumOfTrail) * 1.0))
         : (this.state.numOfFailure + this.state.cachedNumOfFailure);
 
+    LOGGER.info(String.format("Cache Ramp Result : [id = %s, action: %s, %s failure: %d, numOfTrail (%d, %d), numOfSuccess: (%d, %d), numOfFailure: (%d, %d), numOfIgnore: (%d, %d)]"
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java]: Indentation is not correct. If you have follow the formatting guidelines (in Azkaban github page) it should fix it for you.
Also, break thing into multiple lines as per the guidelines.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 2]
```diff
@@ -321,7 +320,7 @@ synchronized void saveSettings() {
         .entrySet()
         .stream()
         .forEach(this::updateExecutedRampFlows);
-    rampDataModel.resetEndFlowCount();
+    rampDataModel.resetFlowCountsAfterSave();
```
Comment (djaiswal83) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: resetFlowCountAfterSave, the "s" is not needed.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 3]
```diff
@@ -401,6 +400,7 @@ synchronized public void configure(ExecutableFlow executableFlow, File flowDirec
                       ExecutableRampStatus.BLACKLISTED.name()
                   )
               );
+              LOGGER.info(String.format("Ramp Flow As BlackListed Item. [rampid = %s, flowName = %s]", rampId, flowName));
```
Comment (djaiswal83) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: Break all these messages into multiple lines as they are too long.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 4]
```diff
@@ -598,43 +602,39 @@ synchronized private Action convertToAction(Status status) {
     // Host the current processing ramp flows
     // Map.Key = FlowId, Map.Value = Set Of Ramps
     private volatile Map<String, Set<String>> executingFlows = new HashMap<>();
-    private volatile int beginFlowCount; // Incoming Ramp Requests After Latest Database Pull Action;
-    private volatile int endFlowCount; // Incoming Ramp Requests After Latest Database Pull Action;
+    private volatile int[] flowCounts = {0, 0}; // [0] count of started flow [1] count of ended flows
```
Comment (djaiswal83) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: The earlier approach looks cleaner and more readable. Why change it to array?

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java [hunk 5]
```diff
@@ -665,8 +665,10 @@ public void start() {
     private void pollExecution() {
       if (this.pollingCriteria.shouldPoll()) {
         if (this.pollingCriteria.satisfiesUnsavedDataAvailableCriteria()) {
+          LOGGER.info("Saving Ramp Setting to Database.");
```
Comment (djaiswal83) [inline azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: For uniformity, either use "Saving and Loading" or "Save and Load". I prefer the latter.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java (hunk 2 above)
Comment (kxu1026) [reply azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: addressing

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java (hunk 5 above)
Comment (kxu1026) [reply azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: good catch. thanks.

File: azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java (hunk 4 above)
Comment (kxu1026) [reply azkaban-exec-server/src/main/java/azkaban/execapp/FlowRampManager.java]: Want to eliminate the potential synchronization problem. I believe volatile int[] way will lock both beginFlowCount and enFlowCount on updating.

**Pull request merged**